src/coffee/directives/api/utils/model-key.coffee

```diff
@@ -135,4 +140,16 @@ angular.module('uiGmapgoogle-maps.directives.api.utils')
           return
         #set isInit to true to force redraw after all updates are processed
         child.updateModel model
+
+
+      modelsLength: =>
+        return 0 unless @scope.models?
+
+        if @scope.models != @cached.models
+          @cached.models = @scope.models
+          array = if angular.isArray(@scope.models) then @scope.models else Object.keys(@scope.models)
+          @cached.modelsLength = array.length
+        #we have existing cachedLength or one was just set
+        return @cached.modelsLength
```

Comment:
Yes the only case this would break is on adding to the model so yeah we would need to check the existing length and or make sure to refresh the length on $watch 'models'